roles/splunk/tasks/remove_splunk.yml

```diff
@@ -11,4 +11,120 @@
+      ansible.builtin.user:
+        name: "{{ splunk_nix_user }}"
+        groups: "{{ splunk_nix_group }}"
+        state: absent
+        local: "{{ local_os_user }}"
+      become: true
+
+    - name: Remove nix splunk group
+      ansible.builtin.group:
+        name: "{{ splunk_nix_group }}"
+        state: absent
+        local: "{{ local_os_group }}"
+      become: true
+  when: splunk_nix_user != 'root'
+
+- name: Remove Disable-THP/Ulimit files
+  block:
+    - name: Stop Disable-THP service
+      ansible.builtin.service:
+        name: disable-thp
+        enabled: false
+        state: stopped
+      become: true
+      when:
+        - ansible_connection!="docker"
+        - "'full' in group_names"
+      ignore_errors: true
+
+    - name: Remove disable-thp service to systemd (CentOS/RHEL 7+)
+      ansible.builtin.file:
+        path: /etc/systemd/system/disable-thp.service
+        state: absent
+      become: true
+      notify:
+        - reload systemctl daemon
+      when:
+        - ansible_os_family == "RedHat"
+        - ansible_distribution_major_version|int > 6
+
+    - name: Remove ulimits
+      ansible.builtin.file:
+        path: /etc/security/limits.d/splunk_ulimits.conf
+        state: absent
+      become: true
+      when: splunk_use_initd
+
+# Not removing DMesg ini config due to defaults on some systems breaking
+
+- name: Remove firewall artifacts
+  block:
+    - name: Remove firewalld artifacts
+      block:
+        - name: Remove splunk firewalld service
+          ansible.builtin.file:
+            path: /etc/firewalld/services/{{ splunk_firewall_service }}.xml
+            state: absent
+          become: true
+
+        - name: Remove firewalld service
+          ansible.posix.firewalld:
+            service: "{{ splunk_firewall_service }}"
+            permanent: true
+            state: enabled
+            immediate: true
+          notify: reload firewalld
+          become: true
+      when: firewall_service == "firewalld"
+
+    - name: Remove UFW ports
+      community.general.ufw:
+        port: "{{ item.number }}"
+        proto: "{{ item.protocol }}"
+        rule: allow
+        state: reloaded
+        comment: "{{ item.desc | default('') }}"
+        delete: true
+      become: true
+      loop: "{{ splunk_firewall_ports }}"
+      when: firewall_service == "ufw"
+  when:
+    - firewall_service != 'undefined'
+    - configure_firewall != false
+    - "'full' in group_names"
+
+- name: Remove FACLs
+  block:
+    - name: Remove setfacl to logrotate script
+      ansible.builtin.lineinfile:
+        path: "{{ logrotate_file }}"
+        line: '        /usr/bin/setfacl -Rm u:{{ splunk_nix_user }}:rx /var/log'
+        state: absent
+      become: true
+
+    - name: Check if auditd.conf is present
+      ansible.builtin.stat:
+        path: /etc/audit/auditd.conf
+      register: result_auditd_conf
+      become: true
+
+    - name: Remove line from /var/log/audit.log
```

Comment:
This may cause a problem because it's removing that line from the conf file. I'm not sure how auditd will behave when that line is not there. The value needs to be set to `root`.

Reply:
That works for me.

Reply:
Fixed in https://github.com/splunk/ansible-role-for-splunk/pull/161/commits/cf45da572ce0537de3d6eb6731700f004a78c51f